include/motor.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef MOTOR_H
+#define MOTOR_H
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
```

Comment:
Should we minimize esp32 dependencies where possible? Raspberry PI might be a target as well if it's serving as an Ethercat Master. Although ESP32 can also be an Ethercat Master